packages/node_modules/pouchdb-replication/src/replicate.js

```diff
@@ -392,6 +392,9 @@ function replicate(src, target, opts, returnValue, result) {
         return;
       }
       return checkpointer.getCheckpoint().then(function (checkpoint) {
+        if (typeof result.initial_checkpoint === 'undefined') {
```

Comment:
I'm not keen on changing the result to support the tests here. I think you can test by intercepting the _changes call as in https://github.com/pouchdb/pouchdb/blob/master/tests/integration/test.replication.js#L2400

Reply:
I agree.  I'll look into the suggested approach.

Reply:
Change and it works, although a little weird as `changes()` is called twice for each call to `replicate()`, first with the initial checkpoint value, and then with the final checkpoint value.